public/app/features/alerting/unified/hooks/useFilteredRules.ts

```diff
@@ -284,6 +323,12 @@ function looseParseMatcher(matcherQuery: string): Matcher | undefined {
   }
 }
 
+// determine which search instance to use, very long search terms should match without checking for Levenshtein distance
+function getSearchInstance(searchTerm: string): uFuzzy {
```

Comment:
You could encapsulate the logic of the `uFuzzy` within here, and do something more like

```ts
const searchTermExeedsMaxNeedleSize = searchTerm.length > MAX_NEEDLE_SIZE;

return new uFuzzy({
  intraMode: searchTermExeedsMaxNeedleSize ? 0 : 1,
  interSplit: '\\s+',
});
```

Then it's clearer what the difference is, without checking the two different `uFuzzy` instances - but this is just a nit pick :)

Reply:
Good point – though since `getSearchInstance` gets called three times (for namespace, group and rule name) we'd be making separate ufuzzy instances. 

I think the performance impact would be negligible though.